src/ChannelServer/World/Entities/Components/Buffs.cs

```diff
@@ -0,0 +1,239 @@
+namespace Melia.Channel.World
+{
+	/// <summary>
+	/// Character buffs.
+	/// </summary>
+	public class Buffs
+	{
+		private readonly Dictionary<BuffId, Buff> _buffs = new Dictionary<BuffId, Buff>();
+
+		/// <summary>
+		/// The owner of this object.
+		/// </summary>
+		public Character Character { get; }
+
+		/// <summary>
+		/// Creates new instance for character.
+		/// </summary>
+		/// <param name="character"></param>
+		public Buffs(Character character)
+		{
+			this.Character = character ?? throw new ArgumentNullException(nameof(character));
+		}
+
+		/// <summary>
+		/// Returns the amount of buffs in the collection.
+		/// </summary>
+		public int Count { get { lock (_buffs) return _buffs.Count; } }
+
+		/// <summary>
+		/// Adds given without updating the client. Replaces existing
+		/// buffs.
+		/// </summary>
+		/// <param name="buff"></param>
+		public void AddSilent(Buff buff)
+		{
+			lock (_buffs)
+				_buffs[buff.Id] = buff;
+		}
+
+		/// <summary>
+		/// Adds given buff and updates the client. Replaces existing
+		/// buffs.
+		/// </summary>
+		/// <param name="buff"></param>
+		public void Add(Buff buff)
+		{
+			var buffExists = _buffs.ContainsKey(buff.Id);
+			this.AddSilent(buff);
+			if (buffExists)
+				Send.ZC_BUFF_UPDATE(this.Character, buff);
+			else
+				Send.ZC_BUFF_ADD(this.Character, buff);
+		}
+
+		/// <summary>
+		/// Removes buff with given id, returns false if it
+		/// didn't exist. Doesn't update the client.
+		/// </summary>
+		/// <param name="buffId"></param>
+		/// <returns></returns>
+		public bool RemoveSilent(BuffId buffId)
+		{
+			lock (_buffs)
+				return _buffs.Remove(buffId);
+		}
+
+		/// <summary>
+		/// Removes buff with given id, returns false if it
+		/// didn't exist. Updates the client on success.
+		/// </summary>
+		/// <param name="buffId"></param>
+		/// <returns></returns>
+		public bool Remove(BuffId buffId)
+		{
+
+			lock (_buffs) {
+				var buff = _buffs[buffId];
+				if (_buffs.Remove(buffId))
+					// Update client...
+					Send.ZC_BUFF_REMOVE(this.Character, buff);
+			}
+
+			return true;
+		}
+
+		/// <summary>
+		/// Removes buff with given id, returns false if it
+		/// didn't exist. Updates the client on success.
+		/// </summary>
+		/// <param name="buff"></param>
+		/// <returns></returns>
+		public bool Remove(Buff buff)
+		{
+			Log.Debug("Removing buff {0} which expired at {1}.", buff.Data.ClassName, buff.RemovalTime);
+			lock (_buffs)
+			{
+				if (!_buffs.Remove(buff.Id))
+					return false;
+				// Update client...
+				Send.ZC_BUFF_REMOVE(this.Character, buff);
+			}
+
+			return true;
+		}
+
+		/// <summary>
+		/// Add a Buff or Update a Buff if it exists
+		/// with a given id.
+		/// </summary>
+		/// <param name="buffId"></param>
+		public void AddOrUpdate(BuffId buffId)
+		{
+			if (!_buffs.ContainsKey(buffId)) {
+				var buff = new Buff(this.Character, buffId);
+				Add(buff);
+			} else
+			{
+				Update(buffId);
+			}
+		}
+
+		/// <summary>
+		/// 
```

Comment:
Empty documentation comments are sad to see. Worse than no comments in a way, because someone took the time to create the structure, but then didn't write anything^^" And a comment might be very helpful for someone, because even if you look at the code, it's not clear at a glance what exactly this method is doing.